### ISSUE TITLE
Consider to add object-property-newline with enhanced object literals?

### ISSUE THREAD

**Issue by Talent30** [eslint-stylistic/eslint-stylistic#41]
Related discussions:
https://github.com/eslint/eslint/issues/6211
https://github.com/eslint/eslint/issues/6251

We would like to accomplish the following use cases by combining `object-property-newline` and `object-curly-newline`
Good
```ts
Object.assign(stream, {
  isTTY: true, columns: false,
});
```
Bad
```ts
Object.assign(stream, {
  isTTY: true, 
  columns: false,
});
```

Good
```ts
Object.assign(stream, {
  isTTY: true,
  columns: false,
  rows: false,
  foo: false,
  bar: false,
  meme: false,
});
```

Bad
```ts
Object.assign(stream, { isTTY: true, columns: false, rows: false, foo: false, bar: false, meme: false, });
```

**Comment by jacekkarczmarczyk** [comment]
I don't understand why 2nd example is bad and 3rd is good

**Comment by Talent30** [comment]
> I don't understand why 2nd example is bad and 3rd is good

There is enough space to display two properties on the same line.

**Comment by jacekkarczmarczyk** [comment]
Would it be possible with this rule to not take line length into account? I wouldn't want my code to be reformatted only because i've added or remove a property

**Comment by Talent30** [comment]
> Would it be possible with this rule to not take line length into account? I wouldn't want my code to be reformatted only because i've added or remove a property

1. It does not consider the line length but instead takes the number of properties into account.
2. This rule exists; however, it is not mandatory to use it. I am merely providing a valid use case to propose an option that offers better control.

**Comment by Talent30** [comment]
Close this issue I will write a better one to refine the use case.
